pkg/cmd/infra/router/template.go

```diff
@@ -79,6 +80,7 @@ func reloadInterval() time.Duration {
 
 func (o *TemplateRouter) Bind(flag *pflag.FlagSet) {
 	flag.StringVar(&o.RouterName, "name", util.Env("ROUTER_SERVICE_NAME", "public"), "The name the router will identify itself with in the route status")
+	flag.StringVar(&o.RouterHostname, "hostName", util.Env("ROUTER_HOSTNAME", ""), "The external host name of the router reported in the route status")
```

Comment:
@fabianofranz I don't think we camelCase flags, right? [oc expose](https://github.com/openshift/origin/blob/9b125cbe4b0ac279087815490778a93188e582e3/pkg/cmd/cli/cmd/expose.go#L70) for example doesn't do this with `hostname`.